geode-redis/src/main/java/org/apache/geode/redis/internal/data/AbstractRedisData.java

```diff
@@ -183,4 +184,20 @@ protected void storeChanges(Region<ByteArrayWrapper, RedisData> region, ByteArra
 
   protected abstract boolean removeFromRegion();
 
+  @Override
+  public boolean equals(Object o) {
+    if (this == o) {
+      return true;
+    }
+    if (!(o instanceof AbstractRedisData)) {
+      return false;
+    }
+    AbstractRedisData that = (AbstractRedisData) o;
+    return getExpirationTimestamp() == that.getExpirationTimestamp();
+  }
+
+  @Override
+  public int hashCode() {
+    return Objects.hash(getExpirationTimestamp());
```

Comment:
is an object's expiration time sufficiently unique to act as a hash?  I assume you've thought through this, but just curious...seems like in a concurrent world, 2 objects of different redistypes could be added concurrently and set to expire at the same time.  Or does our StripedExecutor somehow guard against this?

Reply:
It is but that is not all that is used. That is all we have on AbstractRedisData. Look at its subclasses and you will see they call super.hashCode (this method) and then mix in their own fields to the hash.